### ISSUE TITLE
Rule proposal: `prefer-at` should also mark substrings of length 1

### ISSUE THREAD

**Issue by ST-DDT** [sindresorhus/eslint-plugin-unicorn#2203]
### Description

The `prefer-at` rule should also check for substrings of length 1.

### Fail

```js
string.substring(0, 1)
string.substring(value, value + 1)
string.substring(value, 1 + value)
string.substring(value - 1, value)
```

### Pass

```js
string.at(0)
string.at(value)
string.at(value - 1)
```

**Comment by fisker** [comment]
It's a good idea.

But we have a problem to add this rule.
We already have [`prefer-string-slice`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-slice.md) rule enforce users move to `String#slice()`.

And we can't check `foo.slice(0, 1)`, since `foo` can be an `array` or something else with `.slice()` unless we have type information.

**Comment by sindresorhus** [comment]
While not ideal, the common solution for this is to introduce an option to `prefer-string-slice` to ignore substrings of length 1.

**Comment by sindresorhus** [comment]
Accepted.

**Comment by ST-DDT** [comment]
> While not ideal, the common solution for this is to introduce an option to `prefer-string-slice` to ignore substrings of length 1.

IMO there is no need for a new option, as slice is effectively a substring and thus should also use .charAt/at when possible.

EDIT: Oh I see.
Is there a way to give the prefer-at rule a higher priority than slice? So that it's fixers run first?

Because the option is prone to being configured wrong.
E.g. if it is enabled by default, but prefer-at is not, then it misses changes. If it is not enabled by default, then it is most likely forgotten during initial apply and thus all substrings of length 1 already use slice.
